d9-upgrade.md

```diff
@@ -1,5 +1,61 @@
 # Upgrading nymedia projects to Drupal 9
 
-## Upgrading all of the modules
+## Getting the environment ready
+
+- Make sure the project is using composer 2
+- Make sure the project is using drush 10
+- Make sure the project has at least version ^1.0 of all phpstan related modules (`mglaman/phpstan-drupal`, `phpstan/phpstan-deprecation-rules` and `phpstan/phpstan-phpunit`
+- Make sure the project has upgrade status enabled
+
+## Changing new required settings
+
+You can start by going to `admin/reports/upgrade-status` where this will also be listed. Do the following change to a _committed_ settings.php file, which probably in most project means `drupal/sites/default/settings.php`:
+
+```diff
+--- a/drupal/sites/default/settings.php
++++ b/drupal/sites/default/settings.php
+@@ -790,5 +790,5 @@
+ if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
+   include $app_root . '/' . $site_path . '/settings.local.php';
+ }
+-$config_directories['sync'] = '../config/common';
++$settings['config_sync_directory'] = '../config/common';
+```
+
+## Upgrade all of the contrib modules
+
+It is extremely important to upgrade all of the contrib modules following a specific pattern:
+
+- Upgrade the module with composer
+- Run database updates (if any)
+- Export config (if it differs)
+
+To make this as easy as possible, you can use this one-liner that runs all of those commands and asks for the module name:
+
+``` 
+clear && echo "What is the machine name of the contrib module (i.e commerce, or address)?" && read -r MODULE_NAME && composer update drupal/$MODULE_NAME --with-dependencies && drush updb -y && composer export
```

Comment:
`--with-dependencies` this is redundant on this line?

Reply:
Yes, totally true!